data/src/main/java/com/android254/data/network/util/SafeApiCall.kt

```diff
@@ -43,10 +43,11 @@ class ServerError(cause: Throwable) : Exception(cause)
 class NetworkError : Exception()
 
 suspend fun <T : Any> dataResultSafeApiCall(
-    apiCall: suspend () -> DataResult<T>
+    apiCall: suspend () -> T
 ): DataResult<T> = try {
-    apiCall.invoke()
+    DataResult.Success(apiCall.invoke())
 } catch (throwable: Throwable) {
+//    TODO: This logic doesn't seem right
```

Comment:
Remove TODOs please

Reply:
Okay... but the exception-handling logic on this function needs fixing and I wasn't sure how to fix it. See the last commit 7a87470 for my attempt.

Reply:
You can leave it that way then creat a separate ticket on Trello for fixing that